packages/common-ts/src/base-service/base-service-v2.ts

```diff
@@ -274,17 +274,13 @@ export abstract class BaseServiceV2<
     // names into lower case for the validation step. We'll turn the names back into their original
     // names when we're done.
     const cleaned = cleanEnv<TOptions>(
-      { ...config.env, ...config.args },
+      { ...config.env, ...config.args, ...(params.options || {}) },
       Object.entries(params.optionsSpec || {}).reduce((acc, [key, val]) => {
         acc[key.toLowerCase()] = val.validator({
           desc: val.desc,
           default: val.default,
         })
         return acc
-      }, {}) as any,
-      Object.entries(params.options || {}).reduce((acc, [key, val]) => {
```

Comment:
I misunderstood the interface of `cleanEnv`, the "options" being passed in here are options for the behavior of `cleanEnv` not the actual options that you want to clean. Just added the options field to the first param and it worked.